QualityControl/public/common/object/objectInfoCard.js

```diff
@@ -13,34 +13,93 @@
+const TO_REMOVE_FIELDS = ['etag', 'qcObject', 'versions', 'name', 'location'];
+const HIGHLIGHTED_FIELDS = ['runNumber', 'runType', 'path', 'qcVersion'];
+
+const KEY_TO_RENDER_FIRST = 'path';
 
 /**
  * Builds a panel with information of the object; Fields are parsed according to their category
  * @param {QCObjectDTO} qcObject - QC object with its associated details
  * @param {object} style - properties of the vnode
+ * @param {function(Notification): function(string, string): object} rowAttributes -
+ *  An optional curried function that returns the VNode attribute builder.
+ *  Use {@link defaultRowAttributes} exported from this module, supplying the Notification API.
  * @returns {vnode} - panel with information about the object
+ * @example
+ * ```
+ * qcObjectInfoPanel(qcObject, {}, defaultRowAttributes(model.notification))
+ * ```
  */
-export const qcObjectInfoPanel = (qcObject, style = {}) =>
-  h('.flex-column.scroll-y', { style }, [
-    Object.keys(qcObject)
-      .filter((key) => !TO_REMOVE_FIELDS.includes(key))
-      .map((key) => infoRow(key, qcObject[key])),
+export const qcObjectInfoPanel = (qcObject, style = {}, rowAttributes = () => undefined) =>
+  h('.flex-column.scroll-y#qcObjectInfoPanel', { style }, [
+    [
+      KEY_TO_RENDER_FIRST,
+      ...Object.keys(qcObject)
+        .filter((key) =>
+          key !== KEY_TO_RENDER_FIRST && !TO_REMOVE_FIELDS.includes(key)),
+    ]
+      .map((key) => infoRow(key, qcObject[key], rowAttributes)),
   ]);
 
 /**
  * Builds a raw with the key and value information parsed based on their type
  * @param {string} key - key of the object info
  * @param {string|number|object|undefined} value - value of the object info
+ * @param {function(key, value)} infoRowAttributes - function that return given attributes for the row
  * @returns {vnode} - row with object information key and value
  */
-const infoRow = (key, value) => h('.flex-row.g2', [
-  h('b.w-25.w-wrapped', key),
-  h('.w-75', infoPretty(key, value)),
-]);
+const infoRow = (key, value, infoRowAttributes) => {
+  const highlightedClasses = HIGHLIGHTED_FIELDS.includes(key) ? '.highlighted' : '';
+  const formattedValue = infoPretty(key, value);
+  const formattedKey = getUILabel(key);
+
+  const hasValue = value != null && value !== '' && (!Array.isArray(value) || value.length);
+
+  return h(`.flex-row.g2.info-row${highlightedClasses}`, [
+    h('b.w-25.w-wrapped', formattedKey),
+    h('.w-75', {
+      ...hasValue && infoRowAttributes(formattedKey, formattedValue),
+      style: 'cursor: pointer; user-select: text;',
+    }, formattedValue),
+  ]);
+};
+
+/**
+ * Transforms a camelCase string into human-readable Title Case format,
+ * inserts a space before every uppercase letter and ensures the first
+ * character is capatilized
+ * @param {string} key - key of the object info
+ * @returns {string} - formatted label for the given key
+ */
+const defaultKeyTransform = (key) => {
```

Comment:
Looking at the docs, I believe this method can have a better naming to suggest what it does and be placed in a util file